Bug: Unhandled websocket upgrade request returns 500

### Description

A websocket upgrade request for a path that has no websocket handlers throws an error with a logged stack trace, inclusive of:

```
File ".../litestar/_asgi/routing_trie/traversal.py", line 86, in parse_node_handlers
    return node.asgi_handlers["websocket"]
KeyError: 'websocket'
```

As a user with a logging configuration that downgrades litestar.exception types that are used deliberately to be logged at INFO, this means I'm getting ERROR-level logs going through paths meant for unexpected circumstances.

This can be fixed with:

```patch
diff --git a/litestar/_asgi/routing_trie/traversal.py b/litestar/_asgi/routing_trie/traversal.py
index 1164d0cdb..6742e5c23 100644
--- a/litestar/_asgi/routing_trie/traversal.py
+++ b/litestar/_asgi/routing_trie/traversal.py
@@ -83,7 +83,10 @@ def parse_node_handlers(
             return node.asgi_handlers[method]
         except KeyError as e:
             raise MethodNotAllowedException(headers={"allow": ", ".join(node.asgi_handlers.keys())}) from e
-    return node.asgi_handlers["websocket"]
+    try:
+        return node.asgi_handlers["websocket"]
+    except KeyError as e:
+        raise NotFoundException() from e


 @lru_cache(1024)
```

### URL to code causing the issue

_No response_

### MCVE

```python
import pytest

from litestar import get
from litestar.exceptions import WebSocketDisconnect
from litestar.testing import create_test_client


@get("/")
async def index() -> str:
    return "hello"


def test_websocket_upgrade_to_http_only_route_is_not_found() -> None:
    # litestar's default LoggingConfig dumps a full traceback for every exception it
    # catches, expected or not -- disable it so only a genuinely unexpected failure
    # (the AssertionError below) produces output.
    with create_test_client([index], logging_config=None) as client:
        with pytest.raises(
            WebSocketDisconnect
        ) as unknown_path, client.websocket_connect("/nope"):
            pass
        with pytest.raises(
            WebSocketDisconnect
        ) as http_only_route, client.websocket_connect("/"):
            pass

        assert (
            http_only_route.value.detail == unknown_path.value.detail == "Not Found"
        ), f"{http_only_route.value.detail!r} == {unknown_path.value.detail!r} == 'Not Found'"


if __name__ == "__main__":
    test_websocket_upgrade_to_http_only_route_is_not_found()
    print("PASS")
```

### Steps to reproduce

If the error is present, the MCVE shown above will emit:

```
AssertionError: 'Internal Server Error' == 'Not Found' == 'Not Found'
```

Otherwise, it will emit:

```
PASS
```

### Screenshots

_No response_

### Logs

```text

```

### Litestar Version

2.24.0

### Platform

- [x] Linux
- [x] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

--- a/litestar/_asgi/routing_trie/traversal.py
+++ b/litestar/_asgi/routing_trie/traversal.py
@@ -4,7 +4,7 @@ from functools import lru_cache
 from typing import TYPE_CHECKING, Any
 
 from litestar._asgi.routing_trie.types import PathParameterSentinel
-from litestar.exceptions import MethodNotAllowedException, NotFoundException
+from litestar.exceptions import ClientException, MethodNotAllowedException, NotFoundException
 from litestar.utils import normalize_path
 
 __all__ = ("parse_node_handlers", "parse_path_params", "parse_path_to_route", "traverse_route_map")
@@ -85,7 +85,10 @@ def parse_node_handlers(
             return node.asgi_handlers[method]
         except KeyError as e:
             raise MethodNotAllowedException(headers={"allow": ", ".join(node.asgi_handlers.keys())}) from e
-    return node.asgi_handlers["websocket"]
+    try:
+        return node.asgi_handlers["websocket"]
+    except KeyError as e:
+        raise ClientException() from e
 
 
 @lru_cache(1024)

--- a/tests/unit/test_asgi/test_routing_trie/test_traversal.py
+++ b/tests/unit/test_asgi/test_routing_trie/test_traversal.py
@@ -1,6 +1,9 @@
 from typing import Any
 
+import pytest
+
 from litestar import Router, asgi, get
+from litestar.exceptions import WebSocketDisconnect
 from litestar.response.base import ASGIResponse
 from litestar.status_codes import HTTP_404_NOT_FOUND
 from litestar.testing import create_test_client
@@ -91,3 +94,21 @@ def test_parse_path_to_route_mounted_app_path_router() -> None:
 
         response = client.get("/base/sub/unknown/foobar/")
         assert response.status_code == HTTP_404_NOT_FOUND
+
+
+def test_websocket_upgrade_to_http_only_route_is_bad_request() -> None:
+    # a route that exists but doesn't support the websocket protocol is a client error
+    # (400), distinct from a route that doesn't exist at all (404).
+    @get("/")
+    async def index() -> str:
+        return "hello"
+
+    with (
+        create_test_client([index]) as client,
+        pytest.RaisesGroup(pytest.RaisesExc(WebSocketDisconnect)) as exc_info,
+        client.websocket_connect("/"),
+    ):
+        pass
+
+    assert isinstance(exc_info.value.exceptions[0], WebSocketDisconnect)
+    assert exc_info.value.exceptions[0].detail == "Bad Request"

